tendrl/node_agent/manager/alert_socket.py

```diff
@@ -0,0 +1,88 @@
+import SocketServer
+from multiprocessing import Process
+import json
+from tendrl.node_agent.config import TendrlConfig
+import logging
+from tendrl.node_agent.manager.rpc import EtcdRPC
+from tendrl.common.config import ConfigNotFound
+
+config = TendrlConfig()
+LOG = logging.getLogger(__name__)
+RECEIVE_DATA_SIZE = 4096
+
+
+class AlertUtils():
+    def validate_alert(self, alert_data):
+        alert = json.loads(alert_data)
+        if 'Host' not in alert:
+            raise KeyError('Host')
+        if 'Plugin' not in alert:
+            raise KeyError('Plugin')
+        if 'Severity' not in alert:
+            raise KeyError('Severity')
+        if 'Source' not in alert:
+            raise KeyError('Source')
+        return alert
+
+    def persist_alert(self, alert):
+        key = '/alerts/%s/%s' % (alert['Host'], alert['Plugin'])
+        if 'PluginInstance' in alert:
+            key = "%s/%s" % (key, alert['PluginInstance'])
+        EtcdRPC().client.write(key, alert)
+
+
+class AlertSocketHandler(SocketServer.BaseRequestHandler):
+    def handle(self):
+        try:
+            data = self.request.recv(RECEIVE_DATA_SIZE)
+            alert_utils = AlertUtils()
+            alert = alert_utils.validate_alert(data)
+            alert_utils.persist_alert(alert)
+        except (KeyError, TypeError, ValueError) as ex:
+            LOG.error('Failed to handle data %s on alert socket.Error %s' % (
+                data, ex)
+            )
+        finally:
+            self.request.close()
+
+
+class AlertsServer(SocketServer.ThreadingMixIn, SocketServer.TCPServer):
+    daemon_threads = True
+    allow_reuse_address = True
+
+    def __init__(self, server_address, RequestHandlerClass):
+        SocketServer.TCPServer.__init__(
+            self, server_address, RequestHandlerClass)
+
+
+class AlertsManager(Process):
```

Comment:
Any specific reason to choose Process instead of greenlet eg: https://github.com/Tendrl/node_agent/blob/master/tendrl/node_agent/manager/manager.py#L36